scripts/release.sh

```diff
@@ -0,0 +1,96 @@
+#!/usr/bin/env bash
+#
+# Copyright Kroxylicious Authors.
+#
+# Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+#
+
+set -e
+
+REPOSITORY="origin"
+BRANCH_FROM="main"
+SNAPSHOT_INCREMENT_INDEX=2
+while getopts ":v:b:r:k:n:" opt; do
+  case $opt in
+    v) RELEASE_VERSION="${OPTARG}"
+    ;;
+    b) BRANCH_FROM="${OPTARG}"
+    ;;
+    r) REPOSITORY="${OPTARG}"
+    ;;
+    k) GPG_KEY="${OPTARG}"
+    ;;
+    n) SNAPSHOT_INCREMENT_INDEX="${OPTARG}"
+    ;;
+
+    \?) echo "Invalid option -${OPTARG}" >&2
+    exit 1
+    ;;
+  esac
+
+  case ${OPTARG} in
+    -*) echo "Option $opt needs a valid argument"
+    exit 1
+    ;;
+  esac
+done
+
+if [[ -z "${GPG_KEY}" ]]; then
+    echo "GPG_KEY not set unable to sign the release. Please specify -k <YOUR_GPG_KEY>" 1>&2
+    exit 1
+fi
+
+if [[ -z ${RELEASE_VERSION} ]]; then
+  echo "No version specified aborting"
+  exit 1
+fi
+
+GREEN='\033[0;32m'
+NC='\033[0m' # No Color
+
+git stash --all
```

Comment:
I've normally favoured re-cloning the repo to a temporary location and running the release from there in order to eliminate the risk of pollution from a developer's own workspace.  You are doing a git stash, but what if I have local commits? What do you think?

Reply:
I didn't think too deeply but I thought `stash` plus branch from `remote/branch` would avoid the local commits issue. See [#L55](https://github.com/kroxylicious/kroxylicious-junit5-extension/pull/77/files#diff-1bba00e5aca52286a667c09eff92ba2ca8e3ca77e0d31b0599cc829ab0173389R55) 

Do you have a sample of how you've managed a clean checkout in the past?

Reply:
I don't have any example directly to hand.  I didn't notice the remote/branch.  You approach is good enough.